src/box/memtx_engine.c

```diff
@@ -230,6 +243,8 @@ memtx_engine_recover_snapshot_row(struct memtx_engine *memtx,
 	struct request request;
 	if (xrow_decode_dml(row, &request, dml_request_key_map(row->type)) != 0)
 		return -1;
+	if (not_system_space && request.space_id > BOX_SYSTEM_ID_MAX)
```

Comment:
1. By our code style we should never implicitly convert pointer to bool, we should explicitly compare it with NULL: `not_system_space != NULL`.
2. I don't like that if `request.space_id <= BOX_SYSTEM_ID_MAX` you don't assign the value to false.

Reply:
Fix